src/trace_writer.ml

```diff
@@ -41,19 +41,145 @@ module Thread_info = struct
       mutable callstack : Callstack.t
     ; inactive_callstacks : Callstack.t Stack.t
     ; mutable last_decode_error_time : Time_ns.Span.t
-    ; (* Currently keeping track of the number of frames to unwind during an exception by
```

Comment:
We should keep this workflow and use it when the binary is compiled without the new elf notes.